packages/notebook/src/browser/view-model/notebook-model.ts

```diff
@@ -0,0 +1,129 @@
+// *****************************************************************************
+// Copyright (C) 2023 Red Hat, Inc. and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Disposable, URI } from '@theia/core';
+import { Emitter } from '@theia/core/shared/vscode-languageserver-protocol';
+import { Saveable, SaveOptions } from '@theia/core/lib/browser';
+import { Cell, CellUri, NotebookCellsChangeType, NotebookCellTextModelSplice, NotebookData, NotebookModelWillAddRemoveEvent } from '../../common';
+import { NotebookSerializer } from '../service/notebook-service';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import { NotebookCellModel } from './notebook-cell-model';
+import { MonacoTextModelService } from '@theia/monaco/lib/browser/monaco-text-model-service';
+
+export class NotebookModel implements Saveable, Disposable {
+
+    private readonly dirtyChangedEmitter = new Emitter<void>();
+    readonly onDirtyChanged = this.dirtyChangedEmitter.event;
+
+    private readonly saveEmitter = new Emitter<void>();
+    readonly onDidSaveNotebook = this.saveEmitter.event;
+
+    private readonly didAddRemoveCellEmitter = new Emitter<NotebookModelWillAddRemoveEvent>();
+    readonly onDidAddOrRemoveCell = this.didAddRemoveCellEmitter.event;
+
+    readonly autoSave: 'off' | 'afterDelay' | 'onFocusChange' | 'onWindowChange';
```

Comment:
This property appears to to be unused, can we remove it?

Reply:
for now it's unused but will probably be used in the future. Also it's a requirement for the savable interface